trade-finance-manager-ui/server/controllers/helpers/deal-cancellation-enabled.helper.ff-test.ts

```diff
@@ -0,0 +1,38 @@
+import { DEAL_SUBMISSION_TYPE, TEAM_IDS } from '@ukef/dtfs2-common';
+import { dealCancellationEnabled } from './deal-cancellation-enabled.helper';
+import { TfmSessionUser } from '../../types/tfm-session-user';
+
+const pimUser = { teams: [TEAM_IDS.PIM] } as TfmSessionUser;
+const nonPimUser = { teams: [TEAM_IDS.UNDERWRITERS] } as TfmSessionUser;
+
+describe('dealCancellationEnabled', () => {
+  describe('when `FF_TFM_FACILITY_END_DATE_ENABLED` is set to true', () => {
+    describe('when user is a PIM user', () => {
+      it('should return false if the deal type is MIA', () => {
```

Comment:
Nice assertions could you please add one for an edge case:

* `dealCancellationEnabled({}, {});`
* `dealCancellationEnabled(undefined, undefined);`

Reply:
As we're working in Typescript, this won't work unless we tell the function that we may genuinely expect an undefined or empty object as parameters. If this is genuinely a concern we can change the function to e.g. take `submissionType?: DealSubmissionType | null, user?: TfmSessionUser | null)`: boolean => {`
 but it would be nicer if we don't have to as that would make the function a bit messy. I think elsewhere we generally assume that if the deal exists then the deal submission type exists and so on. What do you think?

Reply:
Ah apologies I thought this was in JS.